db/view/view.cc

```diff
@@ -1307,6 +1307,15 @@ future<query::clustering_row_ranges> calculate_affected_clustering_ranges(const
 
 }
 
+bool needs_static_row(const mutation_partition& mp, const std::vector<view_and_base>& views) {
+    if (!mp.partition_tombstone() && mp.static_row().empty()) {
+        return false;
+    }
+    return std::any_of(views.begin(), views.end(), [] (const view_and_base& vab) {
+        return !vab.base->base_static_columns_in_view_pk().empty();
```

Comment:
This code was removed because it didn't work well after collection indexes were merged and the PR got rebased, but I'll try to explain what was the original motivation.

Yes, all `view_and_base` instances here refer to the same base table. However, they all describe different views for that base table. The purpose of this `std::any_of` was to check whether given base table has at least one index based on one of its static columns. If there are no such indexes then we don't have to fetch existing static rows in read-before-write.